src/App.tsx

```diff
@@ -34,62 +34,112 @@ export function getReorderedGoods(
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;  
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-      <ul>
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByAlphabeth = () => {
+    this.setState({ sortType: SortType.ALPHABET});
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+    const { isReversed, sortType } = this.state;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sortType === SortType.ALPHABET
+              ? 'button is-info' : 'button is-info is-light'}
+            onClick={this.sortByAlphabeth}
+          >
+            Sort alphabetically
+          </button>
+  
+          <button
+            type="button"
+            className={sortType === SortType.LENGTH
+            ? 'button is-success' : 'button is-success is-light'}
+          onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+  
+          <button
+            type="button"
+            className={isReversed
+              ? 'button is-warning' : 'button is-warning is-light'}
+            onClick={this.reverse}  
+          >
+            Reverse
+          </button>
+
+          {(isReversed === true || sortType !== SortType.NONE)
```

Comment:
move to the separate variable
```suggestion
          {...
```